ft-sdk/src/auth/mod.rs

```diff
@@ -120,6 +118,13 @@ pub fn ud(
         .cloned()
         .unwrap_or_else(|| data.emails.first().cloned().unwrap());
 
+    // if the user has no identity, they are not logged in
+    // a user with no identity exists when it is an imported user or some other app has created
+    // user details but they've not done auth stuff (login/signup)
+    if identity.is_none() {
+        return Ok(None);
+    }
```

Comment:
Not happy with this. This is a contract violation. `fastn_session.uid` must only be set for legitimate login, I fear you are putting this logic to handle subscription case. Subscription must put user in session data with some subscription related key, eg `subscription_uid` and not set `fastn_session.uid. Any attempt to set `fastn_session.uid` to a user whose `identity` is null must panic.

Reply:
This makes sense. I'll change the current behavior :+1:

Reply:
this is removed in https://github.com/FifthTry/ft-sdk/pull/52/commits/b43a4475ea28ec94937d6a2d6c466e99302668f3#diff-77d63ae9b1a43d215d7ec0f6831e5b3bdef5d514dcb97075b55f61e6fb110d84L115